samples/monster_generated.h

```diff
@@ -133,6 +136,19 @@ struct EquipmentUnion {
   }
 };
 
+
+inline bool operator==(const EquipmentUnion &lhs, const EquipmentUnion &rhs) {
+  if (lhs.type != rhs.type) return false;
+  switch (lhs.type) {
+    case Equipment_Weapon: {
+      return *(reinterpret_cast<const WeaponT *>(lhs.value)) ==
+             *(reinterpret_cast<const WeaponT *>(rhs.value));
+    }
+    default: {
+      return true;
```

Comment:
This is not safe against future extension, if someone adds a new type to a union, then this will return equal in old code for new values that are not equal. So maybe default false, but add Equipment_None to be true?